src/pages/FarmV3/ActionModal.tsx

```diff
@@ -92,170 +72,153 @@ export const ActionModal: FC<{
-                  height={20}
-                  width={20}
-                />
-              </Tooltip>
-            </div>
-            <div tw="dark:text-grey-5 text-black-4 text-regular font-semibold">
-              {earlyWithdrawFee ? truncateBigNumber(earlyWithdrawFee) + ' ' + token?.token : 0}
+                </IconTooltip>
+              </div>
+              <div className="dark:text-grey-5 text-black-4 text-regular font-semibold">
+                {earlyWithdrawFee ? truncateBigNumber(earlyWithdrawFee) + ' ' + token?.token : 0}
+              </div>
             </div>
-          </div>
-        )}
-        <Button
-          height="35px"
-          cssStyle={tw`duration-500 w-[530px] sm:w-[100%] !h-8.75 bg-blue-1 text-regular border-none mx-auto my-3.75
-                    !text-white font-semibold rounded-[50px] flex items-center justify-center outline-none`}
-          onClick={handleUserAction}
-          loading={isButtonLoading}
-          disabled={
-            (actionType === 'claim' && !claimAmount) ||
-            (actionType === 'deposit' && !depositAmount) ||
-            (actionType === 'withdraw' && !withdrawAmount)
-          }
-          disabledColor={tw`dark:bg-black-1 bg-grey-5 !text-grey-1 opacity-70`}
-        >
-          {`${
-            actionType === 'deposit'
-              ? `Deposit ${truncateBigNumber(+depositAmount)} ${token?.token} + Claim Yield`
-              : actionType === 'withdraw'
-              ? `Withdraw ${truncateBigNumber(+withdrawAmount + claimAmount - earlyWithdrawFee)} ${token?.token}`
-              : `${claimAmount ? `${truncateBigNumber(claimAmount)} ${token?.token}` : '00.00 ' + token?.token}`
-          }`}
-        </Button>
-        <div
-          tw="text-center text-red-2 font-bold text-regular cursor-pointer mb-2.5"
-          onClick={() => setActionModal(false)}
-        >
-          Cancel
+          )}
         </div>
-        <div tw="text-regular dark:text-grey-2 text-grey-1 text-tiny font-semibold text-center">
-          By {actionType === 'withdraw' ? 'withdrawing' : actionType === 'deposit' ? 'depositing' : 'claiming'},
-          you agree to our{' '}
-          <a
-            href="https://www.goosefx.io/terms"
-            target={'_blank'}
-            rel={'noreferrer'}
-            tw="text-tiny font-semibold underline dark:text-white text-blue-1"
+        <div>
+          <Button
+            fullWidth
+            colorScheme={'blue'}
+            className={`duration-500`}
+            onClick={handleUserAction}
+            isLoading={isButtonLoading}
+            disabled={
+              (actionType === 'claim' && !claimAmount) ||
+              (actionType === 'deposit' && !depositAmount) ||
+              (actionType === 'withdraw' && !withdrawAmount)
+            }
           >
-            Terms of Service
-          </a>
-          .
+            {`${
+              actionType === 'deposit'
```

Comment:
It is so difficult to read. Would it be better to make all these checks to separate function?